### PR TITLE
[`v6`] Expose chunk_elements on the ColBERT scorers

### PR DESCRIPTION
**Pull request by tomaarsen**
Hello!

Heads up, this PR was AI-generated and human-reviewed.

## Pull Request overview
* Forward `chunk_elements` through the ColBERT scorers, so the default MVE training path can bound its scoring memory
* Give the `mean_colbert_*` wrappers `length_normalize`, matching `mean_maxsim`
* Rename the private `_document_chunk_ranges` to `_chunk_ranges` and clarify the `chunk_elements` docs

## Details

#3905 gave `maxsim` an element budget and #3931 unified the name to `chunk_elements`, but the `colbert_*` scorers delegate to `maxsim` without forwarding one. Those are the default `similarity_fct` for all four MVE losses, so the default training path was the one place you could not bound the scoring intermediate: `functools.partial(colbert_scores, chunk_elements=N)` raised a `TypeError`, and the only escape was swapping to `XTRScores` or hand-rolling a `partial(maxsim, ...)` with the wrong keyword convention. The niche XTR path had the knob and the default path did not.

All six scorers now take and forward it. Measured on a full training step (forward plus backward, 24 queries by 4 ways by 180 document tokens):

| `chunk_elements` | peak | scores + gradients |
|---|---|---|
| default (100M) | 100.0 MiB | |
| 1_000_000 | 64.7 MiB | bit-identical |
| 200_000 | 54.3 MiB | bit-identical |

It chunks the document axis, so it composes with the loss-level `score_mini_batch_size`, which chunks queries. I added a note to both docstrings saying so, since two differently-named knobs on the same phase is otherwise a guessing game.

The three `mean_colbert_*` wrappers also gain `length_normalize`, defaulting to True the way `mean_maxsim` already does, so the whole family accepts the same keywords and False recovers the plain counterpart.

I deliberately did not expose `maxsim`'s `device`. I had it in at first for full signature parity, but these scorers run inside training losses, where scoring off the embeddings' device is never what you want, and the two evaluators that accept a `similarity_fct` only ever score small candidate sets. There is no measurement that would justify it. A test asserts every scorer rejects `device=`, so the exclusion does not get quietly undone later for the sake of parity.

The rename is mechanical. The private helper packs documents in `maxsim`, query-document pairs in `maxsim_pairwise`, and the flattened `Q * N` document axis in `xtr_scores`, so `document_widths` / `per_document_token` / `per_document_fixed` described only one of its three call sites. Reading `maxsim_pairwise` meant translating "document" to "pair" every time.

The last commit is fallout from reviewing #3931. The IR and NanoBEIR evaluators described `chunk_elements` as budgeting only the 4D scoring intermediate, where `maxsim` documents it as covering the padded documents as well. That matters because `maxsim` also notes that with a small query side the padding, not the score matrix, is the bigger half, which is exactly the evaluator case (NanoBEIR runs about 50 queries against 5000-document chunks), so sizing the budget from the evaluator docs under-provisions. Unifying the name in #3931 also put `chunk_elements` directly next to `corpus_chunk_size` with nothing in either name saying which axis it bounds, so both docstrings now spell that out.

Two test fixes came out of the same review. The `pytest.raises` pattern on the rejection test was `match="chunk_elements"`, which the rename made strictly weaker: it is a substring of `document_chunk_elements`, so a half-finished rename that left the old name in the user-facing error string would still pass. It is anchored now. Nothing exercised the budget on an evaluator at all, so the `partial` was constructed but never called in CI, which is also covered now.

Everything here is v6-only and unreleased (`maxsim` and `multi_vector_encoder` are both absent from v5.7.0), so no deprecation path is owed.

Tests: 1078 passed, 34 skipped across `tests/multi_vector_encoder` and `tests/util`. `ruff check` and `ruff format` clean.

- Tom Aarsen

### CODE REVIEW

File: sentence_transformers/multi_vector_encoder/scoring/colbert.py [hunk 1]
```diff
@@ -46,6 +50,7 @@ def colbert_kd_scores(
                 documents_embeddings[:, j],
                 a_mask=queries_mask,
                 b_mask=documents_mask[:, j] if documents_mask is not None else None,
+                chunk_elements=chunk_elements,
```
[the remark this answers is not in the document]
Comment (tomaarsen) [reply sentence_transformers/multi_vector_encoder/scoring/colbert.py:56]: This one I am not taking, and I want to record why rather than just resolve it.

The mechanism is right: `chunk_elements` took positional slot 5, so `colbert_kd_scores(q, d, q_mask, d_mask, True)` now binds `True` to `chunk_elements` and leaves `length_normalize=False`, which does change scores.

But there are no existing callers to break. The entire `multi_vector_encoder` package is v6-only and unreleased, and so is `maxsim`: neither is present in v5.7.0. Repo-wide there are also zero positional `length_normalize` call sites. A compatibility shim would be dead code on arrival, and type-dispatching on a positional bool would permanently freeze that slot and put an isinstance check in a hot scoring path to protect callers that cannot exist.

The slot order is deliberate. It matches `maxsim` exactly (`a, b, a_mask, b_mask, chunk_elements, length_normalize`), which is the point of the PR: one keyword surface across the family.

The real fix for this class of fragility is keyword-only arguments (a bare `*` before `chunk_elements`), not a shim. That is worth doing across the whole maxsim family at once rather than only here, so I am leaving it out of this PR to avoid diverging colbert from maxsim.
